repo.go

```diff
@@ -470,6 +484,34 @@ func (r *Repo) TimestampWithExpires(expires time.Time) error {
 	return r.setMeta("timestamp.json", timestamp)
 }
 
+func (r *Repo) fileHashes() (map[string]data.Hashes, error) {
+	hashes := make(map[string]data.Hashes)
+	addHashes := func(name string, meta data.Files) {
+		if m, ok := meta[name]; ok {
+			hashes[name] = m.Hashes
+		}
+	}
+	timestamp, err := r.timestamp()
+	if err != nil {
+		return nil, err
+	}
+	snapshot, err := r.snapshot()
```

Comment:
Now that we're doing multiple lookups, does it make sense to memoize these accessors?